### PR TITLE
Feat/add interface

### PR DESCRIPTION
**Pull request by Karenlrx**
feat: 
1. support convert abi struct 
2. refactor client interface
3. remove client privateKey
4. add Makefile and github build.yml

### CODE REVIEW

**Review by Yzyzzzz23 — APPROVED** [review]

File: client.go [hunk 1]
```diff
@@ -1,22 +1,36 @@
...
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 )
 
 type Client interface {
 	Compile(sourceFiles ...string) (*CompileResult, error)
-	Deploy(result *CompileResult, args []interface{}, opts ...Option) ([]string, error)
-	Invoke(contractAbi *abi.ABI, address string, method string, args []interface{}, opts ...TransactionOption) ([]interface{}, error)
+	Deploy(privKey *ecdsa.PrivateKey, result *CompileResult, args []interface{}, opts ...Option) ([]string, error)
+	DeployByCode(privKey *ecdsa.PrivateKey, abi abi.ABI, code string, args []interface{}, opts ...Option) (string, uint64, error)
```
[the remark this answers is not in the document]
Comment (Karenlrx) [reply client.go:16]: chainroll need it

File: rpc.go [hunk 2]
```diff
@@ -190,18 +341,28 @@ func (rpc *EthRPC) EthGasPrice() (*big.Int, error) {
...
-	var err error
+	var (
+		receipt    *types.Receipt
+		err        error
+		otherError error
+	)
 	err = retry.Retry(func(attempt uint) error {
 		receipt, err = rpc.client.TransactionReceipt(context.Background(), hash)
 		if err != nil {
+			if !strings.Contains(err.Error(), ErrorNotFind) {
```
[the remark this answers is not in the document]
Comment (Karenlrx) [reply rpc.go:352]: retry have time limit

File: rpc.go [hunk 3]
```diff
@@ -213,6 +374,25 @@ func (rpc *EthRPC) EthGetTransactionCount(account common.Address, blockNumber *b
 	return nonce, nil
 }
 
+func (rpc *EthRPC) EthGetBlockByNumber(blockNumber *big.Int, excludingTxs bool) (*types.Block, error) {
+	var (
+		err   error
+		block *types.Block
+	)
+	if excludingTxs {
+		blockHeder, err := rpc.client.HeaderByNumber(context.Background(), blockNumber)
```
[the remark this answers is not in the document]
Comment (Karenlrx) [reply rpc.go:383]: eth client  was not exposed to me the bool field，
image

File: config/config.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/viper"
+)
+
+const configName = "bitxhub.toml"
+
+type JsonRpc struct {
+	Addrs     []string `mapstructure:"http_addrs" toml:"http_addrs" json:"http_addrs"`
```
[the remark this answers is not in the document]
Comment (Karenlrx) [reply config/config.go:15]: the tag of "_" must use mapstructure

**Review by jiuhuche120 — APPROVED** [review]

**Pull request merged**